lambda/matching.ml

```diff
@@ -1656,13 +1708,12 @@ let get_key_constant caller = function
 
 let get_args_constant _ rem = rem
 
+let matcher_of_pattern p = fst (Pattern_head.deconstruct p)
```

Comment:
The name of the function seems strange for a function that is just computing a pattern head.

Reply:
We can rename this into `head`. I think the name comes from an intermediary state where this was a partially-applied matching function (so really a pattern-specific matcher). The head now encodes the same information as pure data.

Reply:
@Octachron remarks that the commit message is slightly off (it also comes from this intermediary state that does not exist anymore), the first point of the bullet list describes a change that has already been made in the commit (using `head`). TODO fix.

Reply:
@Octachron I realized that the next commit in the patch stack was fixing the `matcher_of_pattern` issue (by removing the definition and all its uses) and also clarifying the commit message (I misremembered/misunderstood in my message above, there really is something different that can be done to improve this). I pushed this next commit (which obviously we should have included from the start) in the current PR: https://github.com/ocaml/ocaml/pull/9493/commits/c0625a2a558c73ab02d739387cfa45cbdf291675 . If you agree, I propose to *not* edit the commit that introduces `matcher_of_pattern` to rename it, as it is now gone from the final PR state and the intermediary state is reasonably reasonable.

Reply:
(I am going to assume that the "internally coherent" compliment from the other comment box below also applies here, and mark this "resolved".)